codalab/worker/bundle_service_client.py

```diff
@@ -22,27 +22,20 @@ def wrapper(*args, **kwargs):
                 raise BundleServiceException(message + ': ' + str(e), e.client_error)
             except urllib.error.HTTPError as e:
                 try:
-                    client_error = json.loads(e.read().decode())
-
-                    if client_error['error'] == 'invalid_grant':
+                    # Sets the file's current position at the offset 0. Otherwise, e.read() will return an empty string.
+                    e.seek(0)
+                    client_error = e.read()
+                    if e.reason == 'invalid_grant':
                         raise BundleAuthException(
-                            message
-                            + ': '
-                            + http.client.responses[e.code]
-                            + ' - '
-                            + json.dumps(client_error),
+                            message + ': ' + http.client.responses[e.code] + ' - ' + client_error,
```

Comment:
Why is `json.dumps` gone now?

Reply:
1. In the original logic [e.read().decode()](https://github.com/codalab/codalab-worksheets/blob/master/codalab/worker/bundle_service_client.py#L25), `e.read()` was treated as a bytes object that can be decoded. However, in Python3, `e.read()` actually returns a string object which doesn't have `decode()` function. This is where the error `AttributeError: 'str' object has no attribute 'decode'` was coming from. So I removed the `decode()` method.
2. Since `e.read()` will return a plain string in Python3 now, the structure of parsing object `urllib.error.HTTPError` changed quite a lot comparing with the original way. The original `client_error` will return something similar to ```{
    "error": "bad client",
    "msg": "error message"
}```
However, right now, `e.read()` will directly return the error message. I think it might not be necessary to convert it back and forth between a json object and string. (But I am open to discuss this, as one benefit from `json.dumps` is that it can ensure all the characters are ASCII encoded.)